### PR TITLE
[type:optimize] SpringBoot upgrade to 2.6.8

### PR DESCRIPTION
**Pull request by moremind**
// Describe your PR here; eg. optimize spring boot upgrade.
attention：
TODO list as follow:
* 1.springcloud upgrade will support ribbon. shenyu will use shenyu-loadbalancer in future.
* 2.reactor remove elastic and fixed connection pool, please see:https://github.com/reactor/reactor-netty/issues/1499,and this must write with document.
* 3.need optimize files: ShenyuNettyWebServerConfiguration.java
* 4.shenyu-examples to do optimize:
     * 4.1 shenyu-example-grpc can't execute.
     * 4.2 shenyu-example-spring-cloud-dubbo-example can't execute.
     * 4.3 shenyu-example-sofa's example is error.
     * 4.4 shenyu-exmaple-tars's exmaple is error.
* 5. other plugins, I have finish test.

Make sure that:

- [x] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `mvn clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

**Comment by yu199195** [comment]
good job，pls checked ci

**Comment by moremind** [comment]
I will finish it in this week.

**Review by moremind — COMMENTED** [review]
I hava reviewed these files.

File: shenyu-plugin/shenyu-plugin-api/src/main/java/org/apache/shenyu/plugin/api/result/DefaultShenyuEntity.java [hunk 1]
```diff
@@ -32,6 +34,7 @@ public class DefaultShenyuEntity implements Serializable {
 
     private String message;
 
+    @JsonBackReference
```
Comment (moremind) [inline shenyu-plugin/shenyu-plugin-api/src/main/java/org/apache/shenyu/plugin/api/result/DefaultShenyuEntity.java:37]: this is may be cause json `direct self-reference leading to cycle`

File: shenyu-spring-boot-starter/shenyu-spring-boot-starter-plugin/shenyu-spring-boot-starter-plugin-springcloud/src/main/java/org/apache/shenyu/springboot/starter/plugin/springcloud/SpringCloudPluginConfiguration.java [hunk 2]
```diff
@@ -48,6 +52,19 @@
 @ConditionalOnProperty(value = {"shenyu.plugins.spring-cloud.enabled"}, havingValue = "true", matchIfMissing = true)
 public class SpringCloudPluginConfiguration {
 
+
+    /**
+     * custom blocking loadbalancer.
+     * @param loadBalancerClientFactory loadBalancerFactory
+     * @return loadBalancerClient
+     */
+    @ConditionalOnProperty(value = {"spring.cloud.loadbalancer.ribbon.enabled"}, havingValue = "false", matchIfMissing = true)
```
Comment (moremind) [inline shenyu-spring-boot-starter/shenyu-spring-boot-starter-plugin/shenyu-spring-boot-starter-plugin-springcloud/src/main/java/org/apache/shenyu/springboot/starter/plugin/springcloud/SpringCloudPluginConfiguration.java:61]: use spring cloud loadbalancer

File: shenyu-plugin/shenyu-plugin-httpclient/src/main/java/org/apache/shenyu/plugin/httpclient/config/HttpClientProperties.java [hunk 3]
```diff
@@ -789,9 +789,9 @@ public class Ssl {
         private String keyPassword;
 
         /**
-         * The default ssl configuration type. Defaults to TCP.
+         * The default ssl configuration type. Defaults to JDK Provider.
          */
-        private SslProvider.DefaultConfigurationType defaultConfigurationType = SslProvider.DefaultConfigurationType.TCP;
+        private SslProvider defaultConfigurationType = SslProvider.JDK;
```
Comment (moremind) [inline shenyu-plugin/shenyu-plugin-httpclient/src/main/java/org/apache/shenyu/plugin/httpclient/config/HttpClientProperties.java:794]: I can't sure JDK sslProvider or Open_SSL?

File: shenyu-spring-boot-starter/shenyu-spring-boot-starter-gateway/src/main/java/org/apache/shenyu/springboot/starter/netty/ShenyuNettyWebServerConfiguration.java [hunk 4]
```diff
@@ -78,33 +78,17 @@ private static class EventLoopNettyCustomizer implements NettyServerCustomizer {
...
-                            .option(ChannelOption.SO_SNDBUF, nettyTcpProperties.getSocketChannel().getSoSndBuf())
-                            .option(ChannelOption.IP_TOS, nettyTcpProperties.getSocketChannel().getIpTos())
-                            .option(ChannelOption.ALLOW_HALF_CLOSURE, nettyTcpProperties.getSocketChannel().isAllowHalfClosure())
-                            .option(ChannelOption.CONNECT_TIMEOUT_MILLIS, nettyTcpProperties.getSocketChannel().getConnectTimeoutMillis())
-                            .option(ChannelOption.WRITE_BUFFER_WATER_MARK, new WriteBufferWaterMark(nettyTcpProperties.getSocketChannel().getWriteBufferLowWaterMark(),
-                                    nettyTcpProperties.getSocketChannel().getWriteBufferHighWaterMark()))
-                            .option(ChannelOption.WRITE_SPIN_COUNT, nettyTcpProperties.getSocketChannel().getWriteSpinCount())
-                            .option(ChannelOption.AUTO_READ, nettyTcpProperties.getSocketChannel().isAutoRead())
-                            .option(ChannelOption.ALLOCATOR, nettyTcpProperties.getSocketChannel().getAllocator()));
+            // TODO moremind: optimize httpServer
```
Comment (moremind) [inline shenyu-spring-boot-starter/shenyu-spring-boot-starter-gateway/src/main/java/org/apache/shenyu/springboot/starter/netty/ShenyuNettyWebServerConfiguration.java:81]: this http server args need optimize in next pr.

File: shenyu-spring-boot-starter/shenyu-spring-boot-starter-plugin/shenyu-spring-boot-starter-plugin-httpclient/src/main/java/org/apache/shenyu/springboot/starter/plugin/httpclient/HttpClientPluginConfiguration.java [hunk 5]
```diff
@@ -128,50 +134,75 @@ public HttpClient httpClient(final HttpClientProperties properties,
...
+            // TODO moremind add fixed connection pool
+            // reactor remove fixed pool from 0.9.4
+            // reason: https://github.com/reactor/reactor-netty/issues/1499 and https://github.com/reactor/reactor-netty/issues/1960
+            connectionProvider = buildFixedConnectionPool(pool.getName(), pool.getMaxConnections(), pool.getAcquireTimeout(), pool.getMaxIdleTime());
         } else {
-            connectionProvider = ConnectionProvider.elastic(pool.getName(), pool.getMaxIdleTime());
+            // TODO moremind add elastic connection pool
+            // reactor remove elastic pool from 0.9.4
+            // reason: https://github.com/reactor/reactor-netty/issues/1499 and https://github.com/reactor/reactor-netty/issues/1960
+            connectionProvider = buildFixedConnectionPool(pool.getName(), pool.getMaxConnections(), pool.getAcquireTimeout(), pool.getMaxIdleTime());
```
Comment (moremind) [inline shenyu-spring-boot-starter/shenyu-spring-boot-starter-plugin/shenyu-spring-boot-starter-plugin-httpclient/src/main/java/org/apache/shenyu/springboot/starter/plugin/httpclient/HttpClientPluginConfiguration.java:167]: please attention.fix and elastic connection pool need optimize in next released reactor-netty version.now, just support fixed connection pool.

File: shenyu-spring-boot-starter/shenyu-spring-boot-starter-gateway/src/main/java/org/apache/shenyu/springboot/starter/netty/ShenyuNettyWebServerConfiguration.java [hunk 6]
```diff
@@ -78,33 +78,17 @@ private static class EventLoopNettyCustomizer implements NettyServerCustomizer {
...
-                            .selectorOption(ChannelOption.WRITE_BUFFER_WATER_MARK, new WriteBufferWaterMark(nettyTcpProperties.getServerSocketChannel().getWriteBufferLowWaterMark(),
-                                    nettyTcpProperties.getServerSocketChannel().getWriteBufferHighWaterMark()))
-                            .selectorOption(ChannelOption.WRITE_SPIN_COUNT, nettyTcpProperties.getServerSocketChannel().getWriteSpinCount())
-                            .selectorOption(ChannelOption.AUTO_READ, nettyTcpProperties.getServerSocketChannel().isAutoRead())
-                            .selectorOption(ChannelOption.ALLOCATOR, nettyTcpProperties.getServerSocketChannel().getAllocator())
-
-                            .option(ChannelOption.SO_KEEPALIVE, nettyTcpProperties.getSocketChannel().isSoKeepAlive())
-                            .option(ChannelOption.SO_REUSEADDR, nettyTcpProperties.getSocketChannel().isSoReuseAddr())
-                            .option(ChannelOption.SO_LINGER, nettyTcpProperties.getSocketChannel().getSoLinger())
-                            .option(ChannelOption.TCP_NODELAY, nettyTcpProperties.getSocketChannel().isTcpNoDelay())
```
Comment (yu199195) [inline shenyu-spring-boot-starter/shenyu-spring-boot-starter-gateway/src/main/java/org/apache/shenyu/springboot/starter/netty/ShenyuNettyWebServerConfiguration.java:97]: why remove this ？

File: shenyu-spring-boot-starter/shenyu-spring-boot-starter-plugin/shenyu-spring-boot-starter-plugin-httpclient/src/main/java/org/apache/shenyu/springboot/starter/plugin/httpclient/HttpClientPluginConfiguration.java [hunk 7]
```diff
@@ -128,50 +134,75 @@ public HttpClient httpClient(final HttpClientProperties properties,
...
+                    sslContextBuilder.keyManager(ssl.getKeyManagerFactory());
+                    sslContextBuilder.sslProvider(ssl.getDefaultConfigurationType());
+                });
+        sslContextSpec.sslContext(spec)
                 .handshakeTimeout(ssl.getHandshakeTimeout())
                 .closeNotifyFlushTimeout(ssl.getCloseNotifyFlushTimeout())
                 .closeNotifyReadTimeout(ssl.getCloseNotifyReadTimeout());
     }
 
-    private TcpClient setTcpClientProxy(final TcpClient tcpClient, final HttpClientProperties.Proxy proxy) {
```
Comment (yu199195) [inline shenyu-spring-boot-starter/shenyu-spring-boot-starter-plugin/shenyu-spring-boot-starter-plugin-httpclient/src/main/java/org/apache/shenyu/springboot/starter/plugin/httpclient/HttpClientPluginConfiguration.java:146]: why remove this ？

File: shenyu-spring-boot-starter/shenyu-spring-boot-starter-gateway/src/main/java/org/apache/shenyu/springboot/starter/netty/ShenyuNettyWebServerConfiguration.java (hunk 6 above)
Comment (moremind) [reply shenyu-spring-boot-starter/shenyu-spring-boot-starter-gateway/src/main/java/org/apache/shenyu/springboot/starter/netty/ShenyuNettyWebServerConfiguration.java:97]: http server not need so many args,this agrs need optimize with childOption and Option.

Comment (moremind) [reply shenyu-spring-boot-starter/shenyu-spring-boot-starter-gateway/src/main/java/org/apache/shenyu/springboot/starter/netty/ShenyuNettyWebServerConfiguration.java:97]: HttpServer#tcpConfiguration Deprecated Use the other methods exposed by HttpServer to achieve the same configurations. This method will be removed in version 1.1.0.
>>> this from reactor-netty

File: shenyu-spring-boot-starter/shenyu-spring-boot-starter-plugin/shenyu-spring-boot-starter-plugin-httpclient/src/main/java/org/apache/shenyu/springboot/starter/plugin/httpclient/HttpClientPluginConfiguration.java [hunk 8]
```diff
@@ -93,26 +93,32 @@ public HttpClient httpClient(final HttpClientProperties properties,
...
-                    if (Objects.nonNull(loopResources)) {
-                        tcpClient = tcpClient.runOn(loopResources);
-                    }
-                    return tcpClient;
-                });
+                .option(ChannelOption.CONNECT_TIMEOUT_MILLIS, properties.getConnectTimeout());
+        HttpClientProperties.Proxy proxy = properties.getProxy();
+        if (StringUtils.isNotEmpty(proxy.getHost())) {
+            httpClient.proxy(proxySpec -> {
+                ProxyProvider.Builder builder = proxySpec
```
Comment (yu199195) [inline shenyu-spring-boot-starter/shenyu-spring-boot-starter-plugin/shenyu-spring-boot-starter-plugin-httpclient/src/main/java/org/apache/shenyu/springboot/starter/plugin/httpclient/HttpClientPluginConfiguration.java:100]: this maybe Wrapped into a method

Comment (moremind) [reply shenyu-spring-boot-starter/shenyu-spring-boot-starter-plugin/shenyu-spring-boot-starter-plugin-httpclient/src/main/java/org/apache/shenyu/springboot/starter/plugin/httpclient/HttpClientPluginConfiguration.java:100]: ok, I will wrap this method.

**Review by yu199195 — APPROVED** [review]

**Pull request merged**